src/QAT/qat/purr/compiler/execution.py

```diff
@@ -225,6 +229,32 @@ def squash_binary(value):
 
         return results
 
+    def _apply_error_mitigation(self, results, qfile: "QatFile"):
+        """
+        Process any software-driven results transformation, such as taking a raw waveform result
+        and turning it into a bit, or something else.
+        """
+        readout_instructions = [
+            inst for inst in qfile.meta_instructions
+            if isinstance(inst, PostProcessingReadoutMitigation)
+        ]
+        if not readout_instructions:
+            return results
+
+        if len(results) > 1:
+            # TODO: add support for multiple registers
+            raise ValueError(
+                "Cannot have multiple registers in conjunction with readout error mitigation."
+            )
```

Comment:
this check can precede earlier processing to raise early (unless need to use `readout_instructions` when supporting multiple registers)